.github/workflows/test-e2e.yml

```diff
@@ -176,82 +107,40 @@ jobs:
-        if: steps.get-repo-files-from-cache.outputs.cache-hit != 'true'
-        uses: actions/checkout@v4
-        with:
-          ref: ${{github.event.pull_request.head.sha}}
-          submodules: true
-          fetch-depth: 0
-
-      - name: Install devbox
-        uses: jetify-com/devbox-install-action@v0.13.0
+      - uses: jetify-com/devbox-install-action@v0.13.0
         with:
           enable-cache: 'true'
-          
-      - name: Set properties
-        id: properties
-        run: |
-          version=$(echo ${{ matrix.k8s }} | awk -F "-" '{print $1}')
-          platform=$(echo ${{ matrix.k8s }} | awk -F "-" '{print $2}')
-          echo "k8s_version=$version" >> $GITHUB_OUTPUT
-          echo "k8s_platform=$platform" >> $GITHUB_OUTPUT
 
-      - name: Generate configuration for the tests
-        uses: ./.github/actions/gen-install-scripts
+      - uses: ./.github/actions/gen-install-scripts
         with:
-          IMAGE_URL: ${{ env.GHCR_REPO }}:${{ needs.prepare-e2e.outputs.tag }}
-          VERSION: ${{ needs.prepare-e2e.outputs.tag }}
           ENV: dev
+          VERSION: dev
 
-      - name: Change path for the test
+      - name: Extract k8s version/platform
+        id: extract
         run: |
-          awk '{gsub(/cloud.mongodb.com/, "cloud-qa.mongodb.com", $0); print}' bundle/manifests/mongodb-atlas-kubernetes.clusterserviceversion.yaml > tmp && mv tmp bundle/manifests/mongodb-atlas-kubernetes.clusterserviceversion.yaml
+          echo "k8s_version=$(echo '${{ matrix.k8s }}' | awk -F '-' '{print $1}')" >> $GITHUB_OUTPUT
+          echo "k8s_platform=$(echo '${{ matrix.k8s }}' | awk -F '-' '{print $2}')" >> $GITHUB_OUTPUT
 
-      - name: Create k8s Kind Cluster
-        if: ${{ steps.properties.outputs.k8s_platform == 'kind' && !env.ACT }}
+      - name: Setup kind cluster if needed
+        if: ${{ steps.extract.outputs.k8s_platform == 'kind' }}
         uses: helm/kind-action@v1.12.0
         with:
           version: v0.26.0
           config: test/helper/e2e/config/kind.yaml
-          node_image: kindest/node:${{ steps.properties.outputs.k8s_version }}
-          cluster_name: ${{ matrix.test }}
+          node_image: kindest/node:${{ steps.extract.outputs.k8s_version }}
+          cluster_name: ${{ matrix.test }}-${{ matrix.k8s }}
           wait: 180s
 
-      - name: Print kubectl version
-        run: |
-          devbox run -- 'kubectl version'
-
-      - name: Print kubectl version
-        run: |
-          devbox run -- 'kubectl version'
-
-      - name: Install CRDs if needed
-        if: ${{ !( matrix.test == 'helm-update' || matrix.test == 'helm-wide' || matrix.test == 'helm-ns' || matrix.test == 'bundle-test' ) }}
-        run: |
-          devbox run -- 'kubectl apply -f deploy/crds'
-
-      - name: Run e2e test
+      - run: devbox run -- kubectl version
+      - run: devbox run -- kubectl apply -f deploy/crds
+      - run: devbox run -- ./scripts/launch-ci-e2e.sh
```

Comment:
Let's name these steps (and others) that have some relevance in the pipeline, logs will look more descriptive